react-restaurant-review-app/src/components/Register/Register.tsx

```diff
@@ -1,3 +1,56 @@
+import { useState } from "react";
+import { User } from "../../models/user";
+import { post } from "../../utils/service";
+import { useNavigate } from "react-router-dom";
+import styles from "./Register.module.css";
+
 export default function Register() {
-  return <h1>it works!</h1>;
+  const [formData, setFormData] = useState<User>({
+    id: "",
+    userName: "",
+    password: "",
+    role: 2,
+  });
+  const navigate = useNavigate();
+
+  function handleFormChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const { name, value } = event.target;
+
+    setFormData((prevValue) => {
+      return { ...prevValue, [name]: value };
+    });
+  }
+
+  async function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+    let response = await post<User, { id: string }>("register", formData);
+
+    if (response) {
+      navigate("/login");
+    }
+  }
+
+  return (
+    <>
```

Comment:
No need for fragment since we already have a form element that wraps everything else in this component